apps/skilavottord/ws/src/app/modules/recyclingPartner/recyclingPartner.resolver.ts

```diff
@@ -20,62 +28,38 @@ export class RecyclingPartnerResolver {
   async skilavottordAllActiveRecyclingPartners(): Promise<
     RecyclingPartnerModel[]
   > {
-    return await this.recyclingPartnerService.findActive()
+    return this.recyclingPartnerService.findAllActive()
   }
 
-  @Authorize({ roles: [Role.developer, Role.recyclingFund] })
-  @Mutation((_) => Boolean)
+  @Query(() => RecyclingPartnerModel)
+  async skilavottordRecyclingPartner(
+    @Args('input', { type: () => RecyclingPartnerInput })
+    { companyId }: RecyclingPartnerInput,
+  ): Promise<RecyclingPartnerModel> {
+    return this.recyclingPartnerService.findOne(companyId)
+  }
+
+  @Mutation(() => RecyclingPartnerModel)
   async createSkilavottordRecyclingPartner(
-    @Args('companyId') nationalId: string,
-    @Args('companyName') companyName: string,
-    @Args('address') address: string,
-    @Args('postnumber') postnumber: string,
-    @Args('city') city: string,
-    @Args('website') website: string,
-    @Args('phone') phone: string,
-    @Args('active') active: boolean,
+    @Args('input', { type: () => CreateRecyclingPartnerInput })
+    input: CreateRecyclingPartnerInput,
   ) {
-    const rp = new RecyclingPartnerModel()
-    rp.companyId = nationalId
-    rp.companyName = companyName
-    rp.address = address
-    rp.postnumber = postnumber
-    rp.city = city
-    rp.website = website
-    rp.phone = phone
-    rp.active = active
-    logger.info('create new recyclingPartner...' + JSON.stringify(rp, null, 2))
-    await this.recyclingPartnerService.createRecyclingPartner(rp)
-    return true
+    return this.recyclingPartnerService.createRecyclingPartner(input)
```

Comment:
Let's call `this.recyclingPartnerService.findOne` first in order to check if it exists, for better error message.

Reply:
Please read https://github.com/island-is/island.is/pull/6007#discussion_r769735856 first